### PR TITLE
Droth 2504 upgrade frontend libraries

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: UI/src/application.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 (function(application) {
+  var defaultLocation, startupAssetTypeId;
```
Comment (ricardofmsimoes) [inline UI/src/application.js:2]: Won't these variables declared to be global in this way? Is this right?

Comment (RobertVisan) [reply UI/src/application.js:2]: Those variables were already used globally in the JS code, I just declared them initially to avoid grunt errors

Comment (ricardofmsimoes) [reply UI/src/application.js:2]: Understood.

File: UI/src/view/assetSelectionMenu.js [hunk 2]
```diff
@@ -10,9 +10,9 @@
     var pointAssets = $('<div class="point-assets-column"></div>');
     var maintenanceAsset = $('<div class="maintenance-asset-column"></div>');
 
-    var linearGroup = $('<span><h2>' + "Viivamaiset kohteet" + '</h2></span>');
-    var pointGroup = $('<span><h2>' + "Pistemäiset kohteet" + '</h2></span>');
-    var maintenanceGroup = $('<span><h3>' + "Vain ELYn ylläpidossa" + '</h3></span>');
+    var linearGroup = $('<span><h2>"Viivamaiset kohteet</h2></span>');
```
Comment (FilPinto) [inline UI/src/view/assetSelectionMenu.js:13]: this " should be taken out also

File: UI/src/view/linear_asset/manoeuvreStyle.js [hunk 3]
```diff
@@ -48,7 +48,7 @@
             new StyleRule().where('zoomLevel', roadLayer.uiState).is(15).use({ pointRadius: 16 })
         ];
 
-        var defaultStyleProvider = new StyleRuleProvider({stroke: {opacity: 0.65, rotation : '${rotation}'}});
+        var defaultStyleProvider = new StyleRuleProvider({stroke: {opacity: 0.65, rotation : ''}});
```
Comment (FilPinto) [inline UI/src/view/linear_asset/manoeuvreStyle.js:51]: You can take out the rotation completely, the stroke object when entering the StyleRuleProvider will not use rotation, only the icon object

```javascript
var defaultStyleProvider = new StyleRuleProvider({stroke: {opacity: 0.65}});
```

File: UI/src/view/linear_asset/manoeuvreStyle.js [hunk 4]
```diff
@@ -66,7 +66,7 @@
             new StyleRule().where('zoomLevel').is(15).use({ pointRadius: 16 })
         ]);
 
-        var selectionProvider = new StyleRuleProvider({stroke: {opacity: 0.9, rotation : '${rotation}', color: '#00f'},
+        var selectionProvider = new StyleRuleProvider({stroke: {opacity: 0.9, rotation : '', color: '#00f'},
```
Comment (FilPinto) [inline UI/src/view/linear_asset/manoeuvreStyle.js:69]: Same idea as previous comment

File: UI/src/view/point_asset/pointAssetStyle.js [hunk 5]
```diff
@@ -15,15 +15,15 @@
                 icon: {
                     scale : 0.75,
                     rotateWithView: true,
-                    rotation: '${rotation}',
+                    rotation: '',
```
Comment (FilPinto) [inline UI/src/view/point_asset/pointAssetStyle.js:18]: Take out rotation, I think there is no use in this

File: UI/src/view/point_asset/pointAssetStyle.js [hunk 6]
```diff
@@ -15,15 +15,15 @@
...
                     src: 'src/resources/digiroad2/bundle/assetlayer/images/direction-arrow-directional-traffic-sign.svg'
                 }}),
 
             new StyleRule().where('floating').is(true).use({
                 icon: {
                     scale : 0.75,
                     rotateWithView: true,
-                    rotation: '${rotation}',
+                    rotation: '',
```
Comment (FilPinto) [inline UI/src/view/point_asset/pointAssetStyle.js:26]: Take out rotation, I think there is no use in this

File: UI/src/view/assetSelectionMenu.js (hunk 2 above)
Comment (RobertVisan) [reply UI/src/view/assetSelectionMenu.js:13]: yes this should be applied

File: package.json [hunk 7]
```diff
@@ -32,7 +33,7 @@
     "jquery-migrate": "3.3.0",
     "jquery-ui-dist": "1.12.0",
     "jquery-validation": "^1.17.0",
-    "lodash": "4.17.15",
+    "lodash": "4.17.19",
```
Comment (marquesfm) [inline package.json:36]: I think this version will generate problems when they deploy, I think VIITE tried to do that, please stay alert.

Comment (RobertVisan) [reply package.json:36]: This is the versions currently used in VIITE, but I will check on devtest when deployed

File: UI/src/view/linear_asset/manoeuvreStyle.js (hunk 3 above)
Comment (RobertVisan) [reply UI/src/view/linear_asset/manoeuvreStyle.js]: Applied

File: UI/src/view/linear_asset/manoeuvreStyle.js (hunk 4 above)
Comment (RobertVisan) [reply UI/src/view/linear_asset/manoeuvreStyle.js]: Applied

File: UI/src/view/point_asset/pointAssetStyle.js (hunk 5 above)
Comment (RobertVisan) [reply UI/src/view/point_asset/pointAssetStyle.js]: Applied

File: UI/src/view/point_asset/pointAssetStyle.js (hunk 6 above)
Comment (RobertVisan) [reply UI/src/view/point_asset/pointAssetStyle.js]: Applied

**Comment by sasuolanderSito** [comment]
why this branch is still open? There seems to be merge conflict but is there also some other reason why this is still open?

**Comment by sasuolanderSito** [comment]
Päivitys
"eslint": "^5.16.0",
"gruntify-eslint": "^5.0.0",
"lodash": "4.17.19",

lint asetus

**Pull request closed without merging**